src/htm/encoders/SimHashDocumentEncoder.README.md

```diff
@@ -0,0 +1,132 @@
+# SimHash Document Encoder Algorithm Details
+
+SimHash is a Locality-Sensitive Hashing (LSH) algorithm from the world of
+nearest-neighbor document similarity search. It is used by the GoogleBot Web
+Crawler to find near-duplicate web pages.
+
+"Similarity" here refers to bitwise similarity (small hamming distance, high
+overlap), not semantic similarity (encodings for "apple" and "computer" will
+have no relation here). Internally, hamming distances are never considered or
+adjusted -- they're always the result of a kind of dynamic statistical
+distribution.
+
+See Also:
+* Code: `./SimHashDocumentEncoder.hpp`
+* Example: `py/htm/examples/encoders/simhash_document_encoder.py`
+* Tests: `bindings/py/tests/encoders/simhash_document_encoder_test.py`
+
+Learn More:
+* Wikipedia: [Locality-Sensitive Hashing](https://en.wikipedia.org/wiki/Locality-sensitive_hashing)
+* Wikipedia: [SimHash](https://en.wikipedia.org/wiki/SimHash)
+* YouTube: [SimHashing Video](https://www.youtube.com/watch?v=gnraT4N43qo)
+
+
+# Example
+
+> **Note on Hashing:**
+Imagine below that we are using a tiny hashing function which outputs a 9-bit
+long binary string digest. (In reality, we use the SHA3 hashing function, with
+the SHAKE256 XOF to give us a binary string digest output which is the same
+length as our output SDR encoding.)
+
+Let's say you have a document: **["hello", "there", "world"]**.
+
+Tokens:
+
+```
+hello
+there
+world
+```
+
+
+# Parameter `tokenSimilarity` OFF:
+
+### Hash Tokens:
+
+```
+hello => [ 0 1 0 1 0 1 0 1 1 ]
+there => [ 0 0 0 1 0 0 1 0 0 ]
+world => [ 1 0 0 1 0 1 0 1 0 ]
+```
+
+### Prepare Token Hashes for Summation (convert 0 to -1):
+
+```
+hello => [ -1 +1 -1 +1 -1 +1 -1 +1 +1 ]
+there => [ -1 -1 -1 +1 -1 -1 +1 -1 -1 ]
+world => [ +1 -1 -1 +1 -1 +1 -1 +1 -1 ]
+```
+
+### Document Summation:
+
+```
+sums  => [ -1 -1 -3 +3 -3 +1 -1 +1 -1 ]
+```
+
+### Document SimHash (filter >= 0, or choose top N sums for sparse):
```

Comment:
Don't we always use "choos top-N sums" internally? Either `activeBits` or `sparsity` (*size -> active bits)

Reply:
Yes, let me rewrite that a little, I always wanted to describe how a regular non-sparse simhash would work (since that doc is more about the algo).  thx.